CHANGELOG.md

```diff
@@ -7,6 +7,10 @@ this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 ## [Unreleased]
 
+## [0.4.1] - 2023-10-04
+
+Update `pydantic` to ^2.4.2
```

Comment:
This does not follow the format specified by [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).

Reply:
To be more clear, this should go in a `Changed` section (see examples below). It could in my opinion also be good for someone who reads this to also indicate that it's a change from pydantic 1.x to 2.x, not just that it's an update to ^2.4.2 without highlighting the importance of it.